scl/azure/azure-monitor.conf

```diff
@@ -0,0 +1,91 @@
+#############################################################################
+# Copyright (c) 2025 Axoflow
+# Copyright (c) 2025 Tamas Kosztyu <tamas.kosztyu@axoflow.com>
+# Copyright (c) 2025 Attila Szakacs <attila.szakacs@axoflow.com>
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 as published
+# by the Free Software Foundation, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+
+block destination _azure_monitor_internal(
+  dce_uri()
+  dcr_id()
+  table_name()
+  template()
+  auth()
+  ...
+) {
+  http(
+    method("POST")
+    url("`dce_uri`/dataCollectionRules/`dcr_id`/streams/`table_name`?api-version=2023-01-01")
+    headers("Content-Type: application/json")
+    persist_name("azure-monitor,`dce_uri`,`dcr_id`,`table_name`")
+    cloud-auth(
+      `auth`
+    )
```

Comment:
I think we could restrict this to only use the `azure(monitor())` auth, like:
```
    cloud-auth(
      azure(
        monitor(
          `auth`
        )
      )
    )
```

I checked the docs and we probably will never support the other 2 types of auth (Authorization code and Implicit:
https://learn.microsoft.com/en-us/azure/azure-monitor/logs/api/access-api?tabs=rest#request-an-authorization-token